compression/decompression-bad-chunks.tentative.any.js

```diff
@@ -49,34 +49,57 @@ const badChunks = [
   },
 ];
 
+// Test Case Design
+// We need to wait until after we close the writable stream to check if the decoded stream is valid.
+// We can end up in a state where all reads/writes are valid, but upon closing the writable stream an error is detected.
+// (Example: A zlib encoded chunk w/o the checksum).
+
 for (const chunk of badChunks) {
   promise_test(async t => {
     const ds = new DecompressionStream('gzip');
     const reader = ds.readable.getReader();
     const writer = ds.writable.getWriter();
-    const writePromise = writer.write(chunk.value);
-    const readPromise = reader.read();
-    await promise_rejects_js(t, TypeError, writePromise, 'write should reject');
-    await promise_rejects_js(t, TypeError, readPromise, 'read should reject');
+
+    writer.write(chunk.value).then(() => {}, () => {});
+    reader.read().then(() => {}, () => {});
+
+    await promise_rejects_js(t, TypeError, writer.close(), 'write should reject');
+    await promise_rejects_js(t, TypeError, writer.closed, 'write should reject');
+
+    await promise_rejects_js(t, TypeError, reader.read(), 'read should reject');
+    await promise_rejects_js(t, TypeError, reader.closed, 'read should reject');
+
   }, `chunk of type ${chunk.name} should error the stream for gzip`);
 
   promise_test(async t => {
     const ds = new DecompressionStream('deflate');
     const reader = ds.readable.getReader();
     const writer = ds.writable.getWriter();
-    const writePromise = writer.write(chunk.value);
-    const readPromise = reader.read();
-    await promise_rejects_js(t, TypeError, writePromise, 'write should reject');
-    await promise_rejects_js(t, TypeError, readPromise, 'read should reject');
+
+    writer.write(chunk.value).then(() => {}, () => {});
+    reader.read().then(() => {}, () => {});
+
+    await promise_rejects_js(t, TypeError, writer.close(), 'write should reject');
+    await promise_rejects_js(t, TypeError, writer.closed, 'write should reject');
+
+    await promise_rejects_js(t, TypeError, reader.read(), 'read should reject');
+    await promise_rejects_js(t, TypeError, reader.closed, 'read should reject');
```

Comment:
Ditto here 'reader read should reject' and 'reader closed should reject'.
Same for the two other places